docs/proposals/build-skipping.adoc

```diff
@@ -0,0 +1,73 @@
+= Proposal: Multi build repo skipping
+
+== Purpose
+
+Skip builds for which nothing has changed in their working directory.
+
+== Background
+
+For a repo with a single build, the pipeline build will only be triggered if any changes trigger a new pipeline run. 
+
+However if one has multiple builds in one repo, the pipeline build runs all build tasks even for those areas where nothing has changed. 
+
+A simple scenario is a repo with a backend and frontend component whose build outputs are combined by a subsequent docker build in a subsequent deploy task. 
+
+There is no good reason when only the frontend code changes that the backend build is redone as well and vice versa of course.  
+
+== Solution
+
+The solution is only provided for tasks supporting global caching (see separate proposal # 412).
+
+The following is added to the images of these tasks:
+
+- `git`
+
+Task parameter
+
+- `cache-output-dir` a boolean defaulting to true.
+
+
+Task implementations are revised as follows if global caching is enabled in `ods.yaml` (see separate proposal).
+
+The `build-output-cache` is at `$(cache-build-parent-dir)/$(git-sha-working-dir)`, where:
+
+- `cache-build-parent-dir` is a directory dedicated to hold build output from all tasks supporting it. 
+- `git-sha-working-dir` is the git sha of the working directory which is determined as follows: `git-sha-working-dir=$(git rev-parse "HEAD:$working_dir")`  (https://stackoverflow.com/a/58600111).
+
+However `build-output-cache` is not defined if:
+
+- Build caching is not enabled, tasks `cache-output-dir` is set to false in `ods.yaml`. 
+- There is no sufficient free space available as determined in `ods-start` after cleanup.
+- Global caching is not enabled as per its parameters in `ods.yaml` (see proposal #412).
+
+If `build-output-cache` is defined and there is prior build output at that location, cached files are made available in the tasks `output-dir`. The cache and output dir will be on the same PVC so that hard linking is expected to work. 
+In addition the task result set a parameter build-reused to 'true' (or perhaps the git sha). +
+**Question:** can `ods-finish` be changed so that reports of reused builds are not required to be provided?
+
+Otherwise the tasks builds as usual and provides the build output to the `output-dir`.
+
+If `build-output-cache` is defined the task makes the build output available in the the cache. Again this should be speedy as hard linking should enable avoiding to copy these files. In addition this can likely be implemented out side of the individual task build scripts.
+
+**To be determined **: If feasible: script(s) or go programs could be provided that implements the mentioned logic and would be available in all tasks images.
+
+=== Cleanup
+
+`ods-start` is changed to delete prior build output from the cache if it is older than 7 days (for example).
```

Comment:
we could make this a task parameter. for not task params of `ods-start` cannot be customised but this may change in the future. I like 7 days as a default.